analysis/quantifier-comparisons/09-emptydrop-comparisons.Rmd

```diff
@@ -0,0 +1,461 @@
+emptydrops_filter_sces <- emptydrops_filter_sces %>%
+  purrr::map(~ add_cell_mito_qc(.x, mito = mito_genes))
+
+emptydrops_filter_sces_200 <- emptydrops_filter_sces_200 %>%
+  purrr::map(~ add_cell_mito_qc(.x, mito = mito_genes))
+
+emptydrops_filter_sces_500 <- emptydrops_filter_sces_500 %>%
+  purrr::map(~ add_cell_mito_qc(.x, mito = mito_genes))
+
+crlike_filter_sces <- crlike_filter_sces %>%
+  purrr::map(~ add_cell_mito_qc(.x, mito = mito_genes))
+
+knee_filtered_sces <- knee_filtered_sces %>%
+  purrr::map(~ add_cell_mito_qc(.x, mito= mito_genes))
+
+# combine into one list 
+fry_filtered_sces <- list(emptydrops_filter_sces, 
+                          emptydrops_filter_sces_200, 
+                          crlike_filter_sces,
+                          knee_filtered_sces,
+                          emptydrops_filter_sces_500)
+names(fry_filtered_sces) <- c("emptyDrops", 
+                              "emptyDrops200",
+                              "emptyDrops500",
+                              "emptyDropsCellRanger",
+                              "knee")
+
+# convert coldata into dataframe for all alevin-fry filtering methods 
+fry_coldata_df <- purrr::map_df(
+  fry_filtered_sces,
+  ~ purrr::map_df(.x, scpcaTools::coldata_to_df, .id = "quant_id"),
+  .id = "filtering_method"
+) %>%
+  dplyr::mutate(tool = "alevin-fry")
+
+# grab coldata from extra nuclei samples from cellranger 
+nuclei_names <- c("SCPCR000118-cdnapre_mRNA", "SCPCR000119-cdnapre_mRNA")
+nuclei_df <- dplyr::bind_rows(cellranger_nuclei_sces[nuclei_names] %>% 
+purrr::map(scpcaTools::coldata_to_df), .id = "quant_id") %>% 
+  dplyr::mutate(tool = "cellranger")
+
+coldata_cellranger <- dplyr::bind_rows(coldata_cellranger, nuclei_df)
+
+# merge with cellranger coldata
+coldata_cellranger <- coldata_cellranger %>%
+  dplyr::mutate(filtering_method = "CellRanger") %>%
+  dplyr::mutate(quant_id = gsub("-.*", "", quant_id)) %>%
+  dplyr::mutate(quant_id = dplyr::case_when(quant_id == "SCPCR000118" ~ "SCPCR000118-nuclei",
+                                            quant_id == "SCPCR000119" ~ "SCPCR000119-nuclei",
+                                            quant_id == "SCPCR000126" ~ "SCPCR000126-cell",
+                                            quant_id == "SCPCR000127" ~ "SCPCR000127-cell",
+                                            quant_id == "SCPCR000220" ~ "SCPCR000220-nuclei",
+                                            quant_id == "SCPCR000221" ~ "SCPCR000221-nuclei"))
+# rename mito column names to match current version of scpcaTools::add_cell_mito_qc output
+colnames(coldata_cellranger) <- stringr::str_replace(colnames(coldata_cellranger), "^subsets_mito_", "mito_")
+
+# create one coldata df with both alevin fry and cellranger, all filtering methods
+coldata_df <- dplyr::bind_rows(fry_coldata_df, coldata_cellranger)
+```
+
+
+```{r fig.height=10, fig.width=10}
+ggplot(coldata_df, aes(x = tool, y = sum, fill = filtering_method)) + 
```

Comment:
As before, I don't find these plots super helpful: All of the alevin samples are based on the same starting distribution. The median shifts are mostly because we cut off the lower end of the distribution at different points.

That said, there is something here in that the emptyDropsCellRanger seems to have a stricter cutoff at times, with a resulting larger shift at times. So to my eye it is sometimes resulting in some more dramatic shifts (126 & 127, in particular)

Reply:
I just looked back at these, and now I think they don't make sense at least not for 118. For 118, `emptyDropsCellRanger` retains many more cells, which I would expect to result in lower UMI counts on average (there just aren't many high UMI cells!)

Maybe the plots in the html file aren't updated? It looks like there was an error in the code, so these plots and others below migth be old: 
<img width="824" alt="Screen Shot 2021-09-03 at 2 50 42 PM" src="https://user-images.githubusercontent.com/20483/132053224-a0c31721-e786-4cf4-a5e1-5f5ee6c988a4.png">

Reply:
I updated the html to make sure they reflect the correct results, but I actually disagree and think these plots are useful because they show where the cutoffs are and what cells are getting thrown out. We want to make sure we are throwing out cells with low counts, which wasn't necessarily the case with `emptyDrops` `lower=100`. I think with the higher thresholds, we now see a more similar distribution to Cell Ranger, and we are removing cells that we likely don't want to keep in the first place.

Reply:
Thanks for updating the figures: they do look more reasonable now. I still don't think I understand why in the case of 118 the median count decreases with `lower=500`, but I am not too worried about it.

My point with the boxplots is partly just me being ornery about box plots in general. If the distributions are not normal-ish, they tend to present what I think is a misleading picture. In this case there is a lot of wasted ink drawing "outliers" that are identical across most of the comparison distributions, and the upper part of the distributions are irrelevant to the questions we are interested in. What we are really interested in is the bottom of the distribution, and where the cutoffs end up functionally, and that is hard to see from these plots. Ultimately, it doesn't really matter much for this exploration, but I just want to register my reaction.